server/backends/blobstore.go

```diff
@@ -252,3 +260,145 @@ func (g *GCSBlobStore) BlobReader(ctx context.Context, blobName string, offset,
 func (g *GCSBlobStore) BlobWriter(ctx context.Context, blobName string) (io.WriteCloser, error) {
 	return g.bucketHandle.Object(blobName).NewWriter(ctx), nil
 }
+
+// AWS stuff
+type AwsS3BlobStore struct {
+	s3         *s3.S3
+	bucket     *string
+	downloader *s3manager.Downloader
+	uploader   *s3manager.Uploader
+}
+
+func NewAwsS3BlobStore(awsConfig *config.AwsS3Config) (*AwsS3BlobStore, error) {
+	ctx := context.Background()
+
+	var creds *credentials.Credentials
+
+	// See https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html#specifying-credentials
+	if awsConfig.CredentialsProfile != "" {
+		creds = credentials.NewSharedCredentials("", awsConfig.CredentialsProfile)
+	} else {
+		creds = credentials.NewSharedCredentials("", "default")
+	}
+
+	sess, err := session.NewSession(&aws.Config{
+		Region:      aws.String(awsConfig.Region),
+		Credentials: creds,
+	},
+	)
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Create S3 service client
+	svc := s3.New(sess)
+
+	awsBlobStore := &AwsS3BlobStore{
+		s3:         svc,
+		bucket:     aws.String(awsConfig.Bucket),
+		downloader: s3manager.NewDownloader(sess),
+		uploader:   s3manager.NewUploader(sess),
+	}
+
+	err = awsBlobStore.createBucketIfNotExists(ctx, awsConfig.Bucket)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return awsBlobStore, nil
+}
+
+func (a *AwsS3BlobStore) createBucketIfNotExists(ctx context.Context, bucketName string) error {
+	// should to awserr.Error checks via awserr.Error
+	if _, err := a.s3.HeadBucket(&s3.HeadBucketInput{Bucket: aws.String(bucketName)}); err == nil {
+		log.Printf("Creating storage bucket: %s", bucketName)
+		a.s3.CreateBucket(&s3.CreateBucketInput{Bucket: aws.String(bucketName)})
+		err = a.s3.WaitUntilBucketExists(&s3.HeadBucketInput{
+			Bucket: aws.String(bucketName),
+		})
+		return err
+	}
+	return nil
```

Comment:
Don't you have to check `err != nil` or return err so it can be checked elsewhere.

Reply:
I've modified this logic now to reflect AWS's response.  If err != nil, it will check the status code.